app/javascript/admin/controllers/confirmation_modal_controller.js

```diff
@@ -27,8 +40,11 @@ export default class ConfirmationModalController extends ModalController {
       const outcome = await response.json();
 
       if (response.ok) {
-        this.removeBadgeAchievement(itemId);
-        displaySnackbar(outcome.message);
+        this.handleRecord({
```

Comment:
So, this `sendToEndpoint` function is meant to handle as many `destroy` actions in `/admin`.

For some destroy-actions, the record is deleted from a table on the page; in this case, the function needs to remove the record and display a snackbar confirmation.

For other destroy-actions, the record is the page (`show-page` in Rails); for these cases, once the record is deleted, the show-page ceases to exist, so it makes sense to redirect the user back to the list of existing records.

I'm attempting to do these in the `handleRecord` function. For the first use-case (eg `badgeAchievements`), things work well.

However, for cases where we need a redirect (i.e. `window.location.replace`), the redirect seems to be "swallowing" the snackbar confirmation. 🤷🏾‍♀️ 

To solve this, I thought of displaying a Success alert (instead of the snackbar). However, that also gets "swallowed" by the redirect. So I'm not quite sure how to engineer this part of the PR.

HELP! 🙋🏾‍♀️

Reply:
Hmm can you display the snackbar with a message saying: "Record deleted. Redirecting..." and then in 5 seconds redirect? It's not the most elegant solution, but it might work.

Also to clarify, the "snackbar confirmation" _is_ just a confirmation message right? It's not a snackbar where the user has to click "confirm" or something?

Reply:
While `location.replace` does not necessarily stop the code from executing [1], the remaining code will execute on the old page. This means you'll need to find a way to pass this information to the next page, e.g in a query param like `snackbarMessage`. Then you can have an `OnDocumentLoaded` that checks for the presence of this query param and dispatches the event once the new page is done loading.

[1] It may not run at all actually, I just played around a bit and it seems like in most modern browsers will stop after the `replace`, but maybe my test was flawed.

Reply:
@citizen428 Per [Suzanne's observation](https://github.com/forem/forem/pull/14819#discussion_r717727188) on how the snackbar's a11y was negatively affected, I used `localStorage` to pass the outcome massage, instead of through the URL params.

Please re-review 🙏🏾